tests/Microsoft.Identity.Test.Integration/SeleniumTests/InteractiveFlowTests.cs

```diff
@@ -131,26 +132,55 @@ public async Task Interactive_AdfsV4_FederatedAsync()
 
         private async Task RunTestForUserAsync(LabResponse labResponse)
         {
-            Action<IWebDriver> seleniumLogic = (driver) =>
-            {
-                Trace.WriteLine("Starting Selenium automation");
-                driver.PerformLogin(labResponse.User);
-            };
-
-            SeleniumWebUIFactory webUIFactory = new SeleniumWebUIFactory(seleniumLogic, _seleniumTimeout);
-
             PublicClientApplication pca = PublicClientApplicationBuilder.Create(labResponse.AppId)
-                                                                        .WithRedirectUri(SeleniumWebUIFactory.FindFreeLocalhostRedirectUri())
+                                                                        .WithRedirectUri(SeleniumWebUI.FindFreeLocalhostRedirectUri())
                                                                         .BuildConcrete();
 
-            pca.ServiceBundle.PlatformProxy.SetWebUiFactory(webUIFactory);
+            Trace.WriteLine("Part 1 - Acquire a token interactively, no login hint");
+            AuthenticationResult result = await pca
+                .AcquireTokenInteractive(_scopes, null)
+                .WithCustomWebUi(CreateSeleniumCustomWebUI(labResponse.User, false))
+                .ExecuteAsync(new CancellationTokenSource(_interactiveAuthTimeout).Token)
+                .ConfigureAwait(false);
+            IAccount account = await AssertSingleAccountAsync(labResponse, pca, result).ConfigureAwait(false);
+
+            Trace.WriteLine("Part 2 - Clear the cache");
+            await pca.RemoveAsync(account).ConfigureAwait(false);
+            Assert.IsFalse((await pca.GetAccountsAsync().ConfigureAwait(false)).Any());
+
+            Trace.WriteLine("Part 3 - Acquire a token interactively again, with login hint");
+            result = await pca
+                .AcquireTokenInteractive(_scopes, null)
+                .WithCustomWebUi(CreateSeleniumCustomWebUI(labResponse.User, true))
```

Comment:
We recreate a SeleniumCustomWebUI each time. Is it because of selenium constraints?

Reply:
No constraints, we could reuse the same custom web ui, but we'd have to change the selenium logic (first time is without login hint, sencond time is with login hint).